src/camlsnark_c/libsnark-caml/depends/libff/libff/algebra/fields/fp4.hpp

```diff
@@ -67,6 +67,7 @@ class Fp4_model {
     Fp4_model cyclotomic_squared() const;
 
     std::vector<my_Fp> all_base_field_elements() { return { c0.c0, c0.c1, c1.c0, c1.c1 }; }
+    Fp4_model(const std::vector<my_Fp>& v) : c0(my_Fp2(v[0], v[1])), c1(my_Fp2(v[2], v[3])) {};
```

Comment:
Nit: can you move this next to the other `Fp4_model` declarations further up the class.